### PR TITLE
fix(encrypt) crypto parameters for RWX volumes

### PR DESCRIPTION
**Pull request by mantissahz**
Add support for crypto parameters for RWX volumes

Ref: longhorn/longhorn#3353, longhorn/longhorn#4829

### CODE REVIEW

File: controller/share_manager_controller.go [hunk 1]
```diff
@@ -731,21 +733,25 @@ func (c *ShareManagerController) createShareManagerPod(sm *longhorn.ShareManager
...
 		secret, err := c.ds.GetSecretRO(secretRef.Namespace, secretRef.Name)
 		if err != nil {
 			return nil, err
 		}
 
-		cryptoKey = string(secret.Data["CRYPTO_KEY_VALUE"])
+		cryptoKey = string(secret.Data[csi.CryptoKeyValue])
 		if len(cryptoKey) == 0 {
-			return nil, fmt.Errorf("missing CRYPTO_KEY_VALUE in secret for encrypted RWX volume %v", volume.Name)
+			return nil, fmt.Errorf("missing %v in secret for encrypted RWX volume %v", csi.CryptoKeyValue, volume.Name)
```
Comment (innobead) [inline controller/share_manager_controller.go:746]: Should not expose any credentials to log.

File: controller/share_manager_controller.go [hunk 2]
```diff
@@ -731,21 +733,25 @@ func (c *ShareManagerController) createShareManagerPod(sm *longhorn.ShareManager
...
 		}
+		cryptoParams = crypto.NewEncryptParams(
+			string(secret.Data[csi.CryptoKeyProvider]), string(secret.Data[csi.CryptoKeyCipher]),
+			string(secret.Data[csi.CryptoKeyHash]), string(secret.Data[csi.CryptoKeySize]), string(secret.Data[csi.CryptoPBKDF]))
```
Comment (innobead) [inline controller/share_manager_controller.go:750]: Please format this well.

```go
cryptoParams = crypto.NewEncryptParams(
    string(secret.Data[csi.CryptoKeyProvider]), 
    string(secret.Data[csi.CryptoKeyCipher]), 
    string(secret.Data[csi.CryptoKeyHash]), 
    string(secret.Data[csi.CryptoKeySize]), 
    string(secret.Data[csi.CryptoPBKDF])
)
```

**Review by innobead — CHANGES_REQUESTED** [review]
In general, LGTM. Some comments need to resolve.

File: controller/share_manager_controller.go (hunk 1 above)
Comment (mantissahz) [reply controller/share_manager_controller.go:746]: `csi.CryptoKeyValue` is exactly the string `CRYPTO_KEY_VALUE`, I could changed backup if any concerns

File: controller/share_manager_controller.go (hunk 2 above)
Comment (mantissahz) [reply controller/share_manager_controller.go]: Modified. Thanks

**Review by innobead — APPROVED** [review]

**Review by innobead — APPROVED** [review]

**Pull request merged**